src/main/kotlin/cacophony/codegen/instructions/cacophonyInstructions/ControlflowInstructions.kt

```diff
@@ -77,24 +77,24 @@ data class Jz(override val label: BlockLabel) : InstructionTemplates.JccInstruct
 
 data class Jnz(override val label: BlockLabel) : InstructionTemplates.JccInstruction(label, "jnz")
 
-data class Call(
-    val function: Definition.FunctionDeclaration,
-) : Instruction {
-    override val registersRead: Set<Register> =
-        setOf(Register.FixedRegister(HardwareRegister.RSP)).union(
-            SystemVAMD64CallConvention.preservedRegisters().map { Register.FixedRegister(it) },
-        )
-    override val registersWritten: Set<Register> = setOf(Register.FixedRegister(HardwareRegister.RSP))
+data class Call(val function: Definition.FunctionDeclaration) : Instruction {
+    override val registersRead =
+        setOf(Register.FixedRegister(HardwareRegister.RSP)) union
+            SystemVAMD64CallConvention.preservedRegisters().map(Register::FixedRegister)
+    override val registersWritten: Set<Register> =
```

Comment:
I think RSP should not be included in registersWritten? Does it even matter?

Reply:
I'm pretty sure it doesn't matter, since `ret` is always the final instruction. I don't see how adding rsp here changes anything